packages/v8-deopt-viewer/src/index.js

```diff
@@ -110,11 +111,24 @@ export default async function run(srcFile, options) {
 	await mkdir(options.out, { recursive: true });
 
 	console.log("Parsing log...");
-	const logContents = await readFile(logFilePath, "utf8");
+
+	const fd = await openFile(logFilePath);
+	const { buffer: logContentsSlice } = await fd.read({length: 64 * 1024 * 1024});
+	await fd.close();
 
 	// New IC format has 10 values instead of 9
-	const hasNewIcFormat = /\w+IC(,.*){10}/.test(logContents);
-	const rawDeoptInfo = await parseV8Log(logContents, {
+	// todo parse first line - v8-version,8,4,371,19,-node.18,0, instead 
+	// https://github.com/andrewiggins/v8-deopt-viewer/issues/47
+	const hasNewIcFormat = /\w+IC(,.*){10}/.test(logContentsSlice.toString());
+
+	// Error: Cannot create a string longer than 0x1fffffe8 characters
```

Comment:
It's a constant in V8. In Node.js it is exposed as `require('buffer').constants.MAX_STRING_LENGTH`, in Chromium-based browsers it is not accessible from JavaScript AFAIK.

And btw it may change from version to version, and actually did change: in Node.js v12 it was equal to 1_073_741_799, and since then it is 536_870_888.